test/processors.jl

```diff
@@ -40,23 +48,54 @@ end
             @test Dagger.iscompatible(proc, opts, unknown_func, us, 1, us, 2.0)
         end
     end
+    @testset "Function/argument compatability of DaggerKAProc" begin
+        unknown_func = () -> nothing
+        tp = DaggerKAProc(1)
+        op = get_parent(tp)
+        opts = ThunkOptions()
+        us = UnknownStruct()
+        for proc in (op, tp)
+            @test Dagger.iscompatible_func(proc, opts, unknown_func)
+            @test Dagger.iscompatible_arg(proc, opts, us)
+            @test Dagger.iscompatible(proc, opts, unknown_func, us, 1)
+        end
+    value(a) = a
+    @kernel function add(A, B, C)
+        I = @index(Global)
+        C[I] = A[I] + B[I]
+    end
+    A = delayed(value)(ones(512,512))
```

Comment:
We need to be settings the `options` kwarg to `delayed` here to a `ThunkOptions` struct where `proctypes=[DaggerKAProc]`, otherwise these kernels might execute on the wrong kind of processor (specifically, a `ThreadProc`, which won't call `fetch').

Reply:
ok